core/src/test/java/com/linecorp/armeria/client/DnsMetricsTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import java.util.concurrent.ExecutionException;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+import com.linecorp.armeria.common.metric.MoreMeters;
+
+public class DnsMetricsTest {
+
+    @Test
+    void test() throws ExecutionException, InterruptedException {
+
+        final ClientFactory factory = ClientFactory.builder()
+                .meterIdPrefix(new MeterIdPrefix("armeria.dns.metrics.test"))
```

Comment:
When you create a `ClientFactory`, you can specify a `MeterRegistry`. Create a new `MeterRegistry` so that `ClientFactory` does not use the default `MeterRegistry`, then check what's recorded in the `MeterRegistry` you specified. I'd recommend using `PrometheusMeterRegistries.newRegistry()` to create one.

Reply:
Ok.

Reply:
Isn't it just setting a new `MeterRegistry` like
```
final MeterRegistry registry = PrometheusMeterRegistries.newRegistry():
ClientFactory.builder().meterRegistry(registry)...
```
instead of introducing a new `MeterRegirstry`? 🤔

Reply:
Not sure @minwoox. I think the default meter registry in the ClientFactory is used to collect various different stats. See example -
{armeria.client.connections.lifespan#count{protocol=h2c}=0.0, armeria.client.connections.lifespan#total{protocol=h2c}=0.0, armeria.client.connections.lifespan#max{protocol=h2c}=0.0, armeria.client.connections.lifespan#count{protocol=h1c}=0.0, armeria.client.connections.lifespan#total{protocol=h1c}=0.0, armeria.client.connections.lifespan#max{protocol=h1c}=0.0}

Are we ok to reuse the same registry for DNS metrics? If yes then it makes sense. Let me know.

Reply:
Example of DNS metric registry --
{armeria.client.dns.queries.written#count{name=wikipedia.com.,server=/2001:558:feed:0:0:0:0:1}=2.0, armeria.client.dns.queries#count{cause=,name=wikipedia.com.,result=success}=2.0}

Reply:
> Are we ok to reuse the same registry for DNS metrics? If yes then it makes sense. Let me know.

I think so. 😄  We can check whether the metric is recorded correctly or not using the combination of tags. 
https://github.com/line/armeria/blob/master/grpc/src/test/java/com/linecorp/armeria/it/grpc/GrpcMetricsIntegrationTest.java#L201